packages/location-field/src/index.tsx

```diff
@@ -147,6 +151,157 @@ function makeUserOption(userLocation, index, key, activeIndex, selectHandlers) {
   );
 }
 
+const renderFeature = (
+  itemIndex,
+  layerColorMap,
+  feature,
+  operatorIconMap,
+  setLocation,
+  addLocationSearch,
+  showSecondaryLabels,
+  locationSelectedLookup,
+  activeIndex,
+  GeocodedOptionIconComponent,
+  geocoderConfig
+) => {
+  // generate the friendly labels for this feature
+  const { main, secondary } = generateLabel(feature.properties);
+
+  // Create the selection handler
+  const locationSelected = () => {
+    getGeocoder(geocoderConfig)
+      .getLocationFromGeocodedFeature(feature)
+      .then(geocodedLocation => {
+        // add the friendly location labels for use later on
+        geocodedLocation.main = main;
+        geocodedLocation.secondary = secondary;
+        geocodedLocation.name = getCombinedLabel(feature.properties);
+        // Set the current location
+        setLocation(geocodedLocation, "GEOCODE");
+        // Add to the location search history. This is intended to
+        // populate the sessionSearches array.
+        addLocationSearch({ location: geocodedLocation });
+      });
+  };
+
+  // Add to the selection handler lookup (for use in onKeyDown)
+  locationSelectedLookup[itemIndex] = locationSelected;
+
+  // Extract GTFS/POI info and assign to class
+  const { id, layer, secondaryLabels, source } = feature.properties;
+  const classNames = [];
+  let operatorIcon;
+  // Operator only exists on transit features
+  const featureIdComponents = source === "transit" && id.split("::");
+  if (featureIdComponents.length > 1 && featureIdComponents?.[1].length > 0) {
+    const operatorName = featureIdComponents[1]
+      .replace(/ /g, "-")
+      .toLowerCase();
+    classNames.push(`operator-${operatorName}`);
+    operatorIcon = operatorIconMap[operatorName];
+  }
+
+  classNames.push(`source-${source}`);
+  classNames.push(`layer-${layer}`);
+
+  // Create and return the option menu item
+  return (
+    <Option
+      classes={classNames.join(" ")}
+      color={layerColorMap[layer]}
+      icon={operatorIcon || <GeocodedOptionIconComponent feature={feature} />}
+      id={getOptionId(itemIndex)}
+      isActive={itemIndex === activeIndex}
+      key={optionKey++}
+      onClick={locationSelected}
+      title={main}
+      subTitle={secondary}
+      secondaryLabels={secondaryLabels}
+      showSecondaryLabels={showSecondaryLabels}
+    />
+  );
+};
+
+const FeatureHeader = ({
+  title,
+  bgColor,
+  HeaderMessage,
```

Comment:
Use camelCase (`headerMessage`) (it doesn't look like it denotes a component). Multiple instances.